src/Documentation/Advanced-Concepts/Codegen.md

```diff
@@ -0,0 +1,16 @@
+---
+layout: page
+title: Code generation
+---
+
+# Code Generation
+
+**Efficient code generation is one of the pillars of Orleans Runtime**. Orleans Runtime make use of generated code in order to ensure proper serialization of types that are used across the cluster, and generating boilerplate that abstract from the Grain developer the need to understand how method shipping, exception propagation and other internal runtime concepts work. 
```

Comment:
nit: make -> makes

Suggest reword as:

The Orleans Runtime makes use of generated code in order to ensure proper serialization of types that are used across the cluster as well as for generating boilerplate which abstracts away the implementation details of method shipping, exception propagation, and other internal runtime concepts.